cnf-certification-test/accesscontrol/securitycontextcontainer/securitycontextcontainer.go

```diff
@@ -0,0 +1,454 @@
+	var ContainerList []PodListcategory
+	var categoryinfo PodListcategory
+	for j := 0; j < len(containers); j++ {
+		cut := &provider.Container{Podname: podName, Namespace: nameSpace, Container: &containers[j]}
+		percontainerSCC := GetContainerSCC(cut, containerSCC)
+		tnf.ClaimFilePrintf("containerSCC %s is %+v", cut, percontainerSCC)
+		// after building the containerSCC need to check to which category it is
+		categoryinfo = PodListcategory{
+			Containername: cut.Name,
+			Podname:       podName,
+			NameSpace:     nameSpace,
+		}
+		if compareCategory(&Category1, &percontainerSCC, CategoryID1) {
+			tnf.ClaimFilePrintf("Testing if pod belongs to category1 ")
+			categoryinfo.Category = CategoryID1
+		} else if compareCategory(&Category1NoUID0, &percontainerSCC, CategoryID1NoUID0) {
+			tnf.ClaimFilePrintf("Testing if pod belongs to category1NoUID0 ")
+			categoryinfo.Category = CategoryID1NoUID0
+		} else if compareCategory(&Category2, &percontainerSCC, CategoryID2) {
+			categoryinfo.Category = CategoryID2
+		} else if compareCategory(&Category3, &percontainerSCC, CategoryID3) {
+			categoryinfo.Category = CategoryID3
+		} else {
+			categoryinfo.Category = CategoryID4
+		}
+		// after building the containerSCC need to check to which category it is
+		ContainerList = append(ContainerList, categoryinfo)
+	}
+	return ContainerList
+}
+
+func checkContainCategory(addCapability []corev1.Capability, categoryAddCapabilities []string) bool {
+	for _, ncc := range addCapability {
+		return stringhelper.StringInSlice(categoryAddCapabilities, string(ncc), true)
+	}
+	return len(addCapability) > 0
+}
+
+func CheckPod(pod *provider.Pod) []PodListcategory {
+	var containerSCC ContainerSCC
+	containerSCC.HostIPC = NOK
+	if pod.Spec.HostIPC {
+		containerSCC.HostIPC = OK
+	}
+	containerSCC.HostNetwork = NOK
+	if pod.Spec.HostNetwork {
+		containerSCC.HostNetwork = OK
+	}
+	containerSCC.HostPID = NOK
+	if pod.Spec.HostPID {
+		containerSCC.HostPID = OK
+	}
+	containerSCC.SeLinuxContextPresent = NOK
+	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.SELinuxOptions != nil {
+		containerSCC.SeLinuxContextPresent = OK
+	}
+	containerSCC.AllVolumeAllowed, containerSCC.HostDirVolumePluginPresent = AllVolumeAllowed(pod.Spec.Volumes)
+	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.RunAsUser != nil {
+		containerSCC.RunAsUserPresent = OK
+	} else {
+		containerSCC.RunAsUserPresent = NOK
+	}
+	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.FSGroup != nil {
+		containerSCC.FsGroupPresent = OK
+	} else {
+		containerSCC.FsGroupPresent = NOK
+	}
+	return CheckCategory(pod.Spec.Containers, containerSCC, pod.Name, pod.Namespace)
+}
+
+//nolint:gocyclo,funlen
+func compareCategory(refCategory, containerSCC *ContainerSCC, id CategoryID) bool {
+	result := true
+	tnf.ClaimFilePrintf("Testing if pod belongs to category %s", &id)
+	if refCategory.AllVolumeAllowed == containerSCC.AllVolumeAllowed {
```

Comment:
Add comment:
// AllVolumeAllowed reports whether the volumes in the container are compliant to the SCC. True means that all volumes declared in the pod are allowed in the SCC. False means that at least one volume is disallowed